### PR TITLE
BATCHIMP-28: Turn the DBListIdentifierPostprocessor into a generic ListIdentifierPostprocessor

### PR DESCRIPTION
**Pull request by slauriere**
- Lookup keys by values not only for DBLists but for all Lists
- Rename module from 'dblistidentifier' to 'listidentifier'

### CODE REVIEW

File: xwiki-batchimport-postprocessors/listidentifier/src/main/java/org/xwiki/batchimport/postprocessors/ListIdentifierPostprocessor.java [hunk 1]
```diff
@@ -93,18 +93,19 @@ public void postProcessRow(Map<String, String> data, List<String> row, int rowIn
...
                         logger.debug("Value to reverse map: " + dataInFile);
                         // split with separator
                         List<String> dataToStore = new ArrayList<>();
                         for (String v : StringUtils.split(dataInFile, config.getListSeparator())) {
-                            String valueToAdd = values.get(v);
+                            String valueToAdd = lookupKey(values, v, prop);
+
                             if (StringUtils.isNotBlank(valueToAdd)) {
                                 dataToStore.add(valueToAdd);
                             }
```
Comment (lucaa) [inline xwiki-batchimport-postprocessors/listidentifier/src/main/java/org/xwiki/batchimport/postprocessors/ListIdentifierPostprocessor.java:111]: hmmm... I don't see any fallback in lookupKey() to return the value as is for the case when no key can be identified for a given value, and no else branch here either. Is this code reverting the fix that was done for BATCHIMP-26 in the previous pull request?

Comment (slauriere) [reply xwiki-batchimport-postprocessors/listidentifier/src/main/java/org/xwiki/batchimport/postprocessors/ListIdentifierPostprocessor.java:111]: My intent was that this pull request, once merged with the one related to BATCHIMP-26 would provide the expected fallback. I managed to get that result locally by merging the BATCHIMP-26 pull request commit with this one but there might be issues with the merging order, or a conflict (as mentioned by GitHub)? Or I made a mistake? We may re-add the fallback manually if it's easier?

Comment (lucaa) [reply xwiki-batchimport-postprocessors/listidentifier/src/main/java/org/xwiki/batchimport/postprocessors/ListIdentifierPostprocessor.java:111]: Indeed, I guess this is how it should work... However, apparently there are conflicts for this PR. I'll check that and merge.

File: xwiki-batchimport-postprocessors/listidentifier/src/main/java/org/xwiki/batchimport/postprocessors/ListIdentifierPostprocessor.java [hunk 2]
```diff
@@ -119,11 +120,26 @@ public void postProcessRow(Map<String, String> data, List<String> row, int rowIn
...
     }
 
+    protected String lookupKey(Map<String, String> keyByValueMap, String value, PropertyClass prop) {
+        String key = keyByValueMap.get(value);
+
+        if (StringUtils.isBlank(key) && prop instanceof StaticListClass) {
+            // For static lists, case insensitive trimmed value retrieval is attempted
+            // if no key is found for the value as is.
+            for (Map.Entry<String, String> entry : keyByValueMap.entrySet()) {
+                if (entry.getKey().compareToIgnoreCase(value.trim()) == 0) {
```
Comment (lucaa) [inline xwiki-batchimport-postprocessors/listidentifier/src/main/java/org/xwiki/batchimport/postprocessors/ListIdentifierPostprocessor.java:135]: I'd say the entry.getKey() should also be trimmed here, if a proper "normalized" comparison is the purpose, but it's not a big deal if it's not done.

Comment (slauriere) [reply xwiki-batchimport-postprocessors/listidentifier/src/main/java/org/xwiki/batchimport/postprocessors/ListIdentifierPostprocessor.java:135]: Indeed, I made sure to trim the key as well in commit 6223346.

**Pull request merged**